framework/src/mesh/MooseMesh.C

```diff
@@ -738,10 +742,10 @@ MooseMesh::cacheInfo()
   }
 }
 
-std::set<SubdomainID> &
-MooseMesh::getNodeBlockIds(const Node & node)
+const std::set<SubdomainID> &
+MooseMesh::getNodeBlockIds(const Node & node) const
 {
-  return _block_node_list[node.id()];
+  return _block_node_list.at(node.id());
```

Comment:
Of course since it's const, we can't change the `std::map`, so a mooseError is better than an uncaught exception.